pkg/assembler/helper.go

```diff
@@ -18,8 +18,35 @@ package assembler
 import (
 	"reflect"
 	"strings"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
 )
 
+// objectMetadata appends metadata associated with the node
+type objectMetadata struct {
+	// sourceInfo is the file location from which the node was created
+	sourceInfo string
+	// collectorInfo is the collector from which the file that created the node came from
+	collectorInfo string
+}
+
+// NewObjectMetadata creates a new instance to add metadata to nodes
+func NewObjectMetadata(s processor.SourceInformation) *objectMetadata {
```

Comment:
do you think it'd be easier if we just ask for the document itself? do you think that'd be an easier interface? if we do end up storing the information as well, the storage information can be embedded in the document struct as well - thoughts?